static/js/settings_profile_fields.js

```diff
@@ -108,22 +108,24 @@ function clear_form_data() {
     $("#profile_field_choices_row").hide();
 }
 
+function read_field_data_from_form(field_type_id, field_elem) {
+    // Only read field data if we are creating a choice field
+    if (field_type_id === field_types.CHOICE.id) {
+        return read_choice_field_data_from_form(field_elem);
+    }
+}
+
 function create_profile_field(e) {
     e.preventDefault();
     e.stopPropagation();
 
-    var selector = $('.admin-profile-field-form div.choice-row');
     var field_data = {};
     var field_type = $('#profile_field_type').val();
 
-    if (parseInt(field_type, 10) === field_types.CHOICE.id) {
-        // Only read choice data if we are creating a choice field.
-        field_data = read_field_data_from_form(selector);
-    }
-
     var opts = {
         success_continuation: clear_form_data,
     };
+    field_data = read_field_data_from_form(parseInt(field_type, 10), $('.new-profile-field-form'));
```

Comment:
I changed the `.new-profile-field-form` element here to not fetch from `e` because there will be only one element anyway in form. It should not create any race condition or other errors. It should work, I tested it manually as well as in casper tests.